sdk/core/azure-core/src/http/bearer_token_authentication_policy.cpp

```diff
@@ -13,6 +13,7 @@ using Azure::Core::Context;
 using Azure::Core::Credentials::AuthenticationException;
 using Azure::Core::Credentials::TokenRequestContext;
 using Azure::Core::Credentials::_detail::AuthorizationChallengeHelper;
+using Azure::Core::Http::Request;
```

Comment:
Is this line necessary? If so, why?

Reply:
It's using that definition e.g. in the signature of  BearerTokenAuthenticationPolicy::AuthenticateAndAuthorizeRequest():

> void BearerTokenAuthenticationPolicy::AuthenticateAndAuthorizeRequest(
>     Request& request,  <---- here
>     Credentials::TokenRequestContext const& tokenRequestContext,
>     TokenRequestContext const& tokenRequestContext,
>     Context const& context) const

Without it I get these errors:

> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(20): error C2065: 'Request': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(20): error C2065: 'request': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(21): error C2275: 'Azure::Core::Http::Policies::NextHttpPolicy': illegal use of this type as an expression
>   C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\inc\azure/core/http/policies/policy.hpp(303): note: see declaration of 'Azure::Core::Http::Policies::NextHttpPolicy'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(21): error C2146: syntax error: missing ')' before identifier 'nextPolicy'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(23): error C2143: syntax error: missing ';' before '{'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(23): error C2447: '{': missing function header (old-style formal list?)
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(44): error C2065: 'Request': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(44): error C2065: 'request': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(45): error C2065: 'nextPolicy': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(45): error C2275: 'Azure::Core::Http::Policies::NextHttpPolicy': illegal use of this type as an expression
>   C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\inc\azure/core/http/policies/policy.hpp(303): note: see declaration of 'Azure::Core::Http::Policies::NextHttpPolicy'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(46): error C2275: 'Azure::Core::Context': illegal use of this type as an expression
>   C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\inc\azure/core/context.hpp(46): note: see declaration of 'Azure::Core::Context'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(46): error C2059: syntax error: 'const'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(47): error C2143: syntax error: missing ';' before '{'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(47): error C2447: '{': missing function header (old-style formal list?)
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(54): error C2061: syntax error: identifier 'Request'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(56): error C2511: 'bool Azure::Core::Http::Policies::_internal::BearerTokenAuthenticationPolicy::AuthorizeRequestOnChallenge(const std::string &) const': overloaded member function not found in 'Azure::Core::Http::Policies::_internal::BearerTokenAuthenticationPolicy'
>   C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\inc\azure/core/http/policies/policy.hpp(546): note: see declaration of 'Azure::Core::Http::Policies::_internal::BearerTokenAuthenticationPolicy'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(58): error C2065: 'request': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(59): error C2065: 'context': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(65): error C2065: 'Request': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(65): error C2065: 'request': undeclared identifier
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(66): error C2275: 'Azure::Core::Credentials::TokenRequestContext': illegal use of this type as an expression
>   C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\inc\azure/core/credentials/credentials.hpp(44): note: see declaration of 'Azure::Core::Credentials::TokenRequestContext'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(66): error C2059: syntax error: 'const'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(68): error C2143: syntax error: missing ';' before '{'
> C:\Project\open-vds\3rdparty\azure-sdk-for-cpp-1.10.3\sdk\core\azure-core\src\http\bearer_token_authentication_policy.cpp(68): error C2447: '{': missing function header (old-style formal list?)
>